tests/components/abode/test_init.py

```diff
@@ -0,0 +1,41 @@
+"""Tests for the Abode module."""
+from unittest.mock import patch
+
+from homeassistant.components import abode
+from homeassistant.components.alarm_control_panel import DOMAIN as ALARM_DOMAIN
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from .common import setup_platform
+
+
+async def test_abode_setup_from_config(hass, requests_mock):
+    """Test setup from configuration yaml file."""
+    config = {
+        abode.DOMAIN: {CONF_USERNAME: "foo", CONF_PASSWORD: "bar", "polling": True}
+    }
+    response = await abode.async_setup(hass, config)
```

Comment:
We want to avoid interacting directly with integration code. Doing so will make the tests more robust.

Use `async_setup_component` instead.